lab_1/main.py

```diff
@@ -4,14 +4,23 @@
 """
 
 
+import re
+
+
 def tokenize(text: str) -> list or None:
     """
     Splits a text into tokens, converts the tokens into lowercase,
     removes punctuation and other symbols from words
     :param text: a text
     :return: a list of lower-cased tokens without punctuation
     """
-    pass
+    if not isinstance(text, str):
+        return None
+    else:
+        text = text.lower()
+        text = re.sub(r'[^\w\s]', '', text)
```

Comment:
what happens on this line?

Reply:
r’[^\w\s] is a regular expression pattern for choosing symbols and numbers; function re.sub checks for symbols, not matching for that pattern and removes them (in that case all symbols except letters will be removed)